python/tank/bootstrap/manager.py

```diff
@@ -221,11 +244,50 @@ def set_progress_callback(self, callback):
         """
         self._progress_cb = callback
 
-    def bootstrap_engine(self, engine_name, entity=None):
+    def bootstrap_engine(self,
```

Comment:
We actually already have a property for [progress callback](http://developer.shotgunsoftware.com/tk-core/bootstrap.html#sgtk.bootstrap.ToolkitManager.set_progress_callback) - any reason we are not using that?

I was imagining that the `bootstrap_engine()` signature wouldn't change - that we would either use QT signals or extend the `set_progress_callback()`.

Given how the implementation looks, I agree with @josh-t  - I think we have two options:

- add QT signals for completion and failure (the original sketchy suggestion for the ticket)
- use separate methods for bootstrapping sync and async.

I am inclined to think it's better for us to keep the QT dependency internal for the time being, and hence implement the following:

```
# existing method is unchanged
def bootstrap_engine(self, engine_name, entity=None)

# we drop the async property and introduce a new method
def bootstrap_engine_async(self, engine_name, entity=None, fail_cb=None, complete_cb=None)
```

The existing `progress_callback` is used in both cases.

What do you guys think?